mx.c

```diff
@@ -1671,19 +1674,21 @@ struct Mailbox *mx_mbox_find2(const char *path)
   if (!path)
     return NULL;
 
-  char buf[PATH_MAX] = { 0 };
-  mutt_str_copy(buf, path, sizeof(buf));
+  struct Buffer *buf = buf_new(path);
   const char *const c_folder = cs_subset_string(NeoMutt->sub, "folder");
-  mx_path_canon(buf, sizeof(buf), c_folder, NULL);
+  mx_path_canon(buf, c_folder, NULL);
 
   struct Account *np = NULL;
   TAILQ_FOREACH(np, &NeoMutt->accounts, entries)
   {
-    struct Mailbox *m = mx_mbox_find(np, buf);
-    if (m)
+    struct Mailbox *m = mx_mbox_find(np, buf_string(buf));
+    if (m) {
+      buf_pool_release(&buf);
```

Comment:
can I use `buf_pool_release()` with a Buffer I didn't get with `buf_pool_get()`? Or should I use `buf_free()` here and in line 1691?

Reply:
No, you'll need to `buf_free()` them.

The Pool is only good for short-term Buffers.
It keeps a cache and recycles Buffers on `buf_pool_release()` -- it normalises their size and zeros them.